daemon/cmd/daemon.go

```diff
@@ -555,7 +555,7 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup, params *daemonParams
 			k := key.(*ipcachemap.Key)
 			v := value.(*ipcachemap.RemoteEndpointInfo)
 			nid := identity.NumericIdentity(v.SecurityIdentity)
-			if nid.HasLocalScope() {
+			if scope := nid.Scope(); scope == identity.IdentityScopeLocalCIDR || scope == identity.IdentityScopeRemoteNode {
```

Comment:
If someone installs with this PR and hence populates the ipcache with identities in the new range, then turns the flag off, I guess that means that this function will still allocate those previous IdentityScopeRemoteNode identities on startup here, then later the agent will release them, and they'll be unused right?

Just wondering whether we need to restore the `IdentityScopeRemoteNode` identities if the flag is off?

Reply:
We will not actually allocate those identities (because the allocator will reject them, since the `ScopeForLabels()` will mismatch. However, it's safer to short-circuit here. I'll add that.

Reply:
Done.